components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2125,9 +2127,28 @@ where
         if meta.region_ranges.remove(&last_key).is_none() {
             panic!("{} original region should exists", self.fsm.peer.tag);
         }
-        // It's not correct anymore, so set it to None to let split checker update it.
-        self.fsm.peer.approximate_size = None;
         let last_region_id = regions.last().unwrap().get_id();
+
+        let new_region_count = regions.len() as u64;
+        // Roughly estimate the size and keys and then let split checker to update it.
+        self.fsm.peer.approximate_size =
+            self.fsm.peer.approximate_size.map(|x| x / new_region_count);
```

Comment:
When Region [a,e) size exceeds `region_max_size`, it will be split into several Regions [a,b), [b,c), [c,d), [d,e) and the size of [a,b), [b,c), [c,d) will be `region_split_size` (or a little larger)

Let's make the this peer's approximate_size to be `x - (region_split_size * (n -1))` and new region peer's to be `region_split_size`

Reply:
It seems this way is more reasonable. And what about the keys calculation?

Reply:
I found it's hard to access `region_split_size` from current peer's member variables. Since the keys and size will be updated by split checker very soon, finally I keep the original implementation.